### PR TITLE
feat(api) new endpoint : FindGraphTemplates

### PR DESCRIPTION
**Pull request by a-launois**
## Description

new configuration endpoint:
```
GET /configuration/graphs/templates
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)
- [x] 24.04.x

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/php/Core/GraphTemplate/Domain/Model/GraphTemplateTest.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    \Assert\InvalidArgumentException::class,
+    AssertionException::maxLength(
+        str_repeat('a', GraphTemplate::LABEL_MAX_LENGTH + 1),
+        GraphTemplate::LABEL_MAX_LENGTH + 1,
+        GraphTemplate::LABEL_MAX_LENGTH,
+        'GraphTemplate::verticalAxisLabel'
+    )->getMessage()
+);
+
+it('should throw an exception when base is different than 1000 or 1024', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/GraphTemplate/Domain/Model/GraphTemplateTest.php:148]:
```suggestion
it('should throw an exception when base is different from 1000 or 1024', function (): void {
```

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    {
+        return $this->height;
+    }
+
+    public function getBase():  int
+    {
+        return $this->base;
+    }
+
+    public function getGridLowerLimit():  ?float
```
Comment (abuathier) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:108]: Double space after colon everywhere in this file.
```suggestion
    public function getGridLowerLimit(): ?float
```

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        Assertion::notEmptyString($name, 'GraphTemplate::name');
+        Assertion::maxLength($name, self::NAME_MAX_LENGTH, 'GraphTemplate::name');
+
+        Assertion::notEmptyString($verticalAxisLabel, 'GraphTemplate::verticalAxisLabel');
+        Assertion::maxLength($verticalAxisLabel, self::LABEL_MAX_LENGTH, 'GraphTemplate::verticalAxisLabel');
+
+        Assertion::inArray($base, self::BASE_ALLOWED_VALUES, 'GraphTemplate::base');
+
+        if ($this->isUpperLimitSizedToMax === true) {
```
Comment (abuathier) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:73]: This is a hard typed boolean, strictness with `true` is unnecessary.
```suggestion
        if ($this->isUpperLimitSizedToMax) {
```

Comment (abuathier) [reply centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:73]: Addendum : Is it a domain failure (inconsistency) if we set `gridUpperLimit` to `true|false` while having `isUpperLimitSizedToMax` set to `true` ?
- if yes, check the case and throw an assert exception
- if no, please add a comment which tells explicitely that we ignore such cases (quietly)

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        private string $name,
+        private string $verticalAxisLabel,
+        private int $width,
+        private int $height,
+        private int $base = 1000,
+        private ?float $gridLowerLimit = null,
+        private ?float $gridUpperLimit = null,
+        private bool $isUpperLimitSizedToMax = false,
+        private bool $isGraphScaled = false,
+        private bool $isDefaultCentreonTemplate = false,
```
Comment (abuathier) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:61]: This class has no setter, you can pass the parameters **`readonly`**.

**Review by dmyios — APPROVED** [review]

File: centreon/doc/API/v23.10/Configuration/GraphTemplate/Schema/FindGraphTemplatesResponse.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    example: 115
+  gridUpperLimit:
+    type: float
+    nullable: true
+    description: "Upper limit of the grid"
+    example: 115
+  isUpperLimitSizedToMax:
+    type: bool
+    description: "Set the upper limit of the grid to the maximum value"
+    example: false
```
Comment (callapa) [inline centreon/doc/API/v23.10/Configuration/GraphTemplate/Schema/FindGraphTemplatesResponse.yaml:40]: The documentation does not correspond to the presenter.

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+        Assertion::notEmptyString($verticalAxisLabel, 'GraphTemplate::verticalAxisLabel');
+        Assertion::maxLength($verticalAxisLabel, self::LABEL_MAX_LENGTH, 'GraphTemplate::verticalAxisLabel');
+
+        Assertion::inArray($base, self::BASE_ALLOWED_VALUES, 'GraphTemplate::base');
+
+        // if isUpperLimitSizedToMax is set to true, gridUpperLimit is silently set to null
+        if ($this->isUpperLimitSizedToMax) {
+            $this->gridUpperLimit = null;
+        }
```
Comment (callapa) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:76]: And if gridUpperLimit is set to something not null, is isUpperLimitSizedToMax set to false ?

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        private int $base = 1000,
+        private ?float $gridLowerLimit = null,
+        private ?float $gridUpperLimit = null,
+        private bool $isUpperLimitSizedToMax = false,
+        private bool $isGraphScaled = false,
+        private bool $isDefaultCentreonTemplate = false,
+    ) {
+        Assertion::positiveInt($id, 'GraphTemplate::id');
+
+        Assertion::notEmptyString($name, 'GraphTemplate::name');
```
Comment (callapa) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:65]: I think you've forgotten about the "trim" function before

File: centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        Assertion::maxLength($name, self::NAME_MAX_LENGTH, 'GraphTemplate::name');
+
+        Assertion::notEmptyString($verticalAxisLabel, 'GraphTemplate::verticalAxisLabel');
```
Comment (callapa) [inline centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php:68]: I think you've forgotten about the "trim" function before

**Review by callapa — APPROVED** [review]

**Review by cg-tw — DISMISSED** [review]

**Review by arnapou — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**